lib/srv/desktop/audit.go

```diff
@@ -23,10 +23,43 @@ import (
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/api/types/events"
 	libevents "github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/generics"
+	"github.com/gravitational/teleport/lib/srv/desktop/tdp"
 	"github.com/gravitational/teleport/lib/tlsca"
 	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
 )
 
+const (
+	unknownName = "unknown"
+)
+
+// sharedDirectoryNameMap keeps a mapping of sessionId
+// to a shared directory's name, for use by the audit log.
+type sharedDirectoryNameMap struct {
```

Comment:
This assumes only 1 shared directory per session, right?

Do you think we'll ever support sharing > 1 directory?

Reply:
> This assumes only 1 shared directory per session, right?

It does. 

> Do you think we'll ever support sharing > 1 directory?

Yeah, I should just implement it to support multiple directories now.